code/game/gamemodes/nuclear/pinpointer.dm

```diff
@@ -87,32 +141,23 @@
 /obj/item/pinpointer/advpinpointer
 	name = "advanced pinpointer"
 	desc = "A larger version of the normal pinpointer, this unit features a helpful quantum entanglement detection system to locate various objects that do not broadcast a locator signal."
-	var/mode = 0  // Mode 0 locates disk, mode 1 locates coordinates.
+	modes = list(MODE_ADV)
 	var/modelocked = FALSE // If true, user cannot change mode.
 	var/turf/location = null
 	var/obj/target = null
+	var/setting = 0
 
-/obj/item/pinpointer/advpinpointer/attack_self()
-	if(!active)
-		active = 1
-		if(mode == 0)
-			workdisk()
-		if(mode == 1)
-			point_at(location)
-		if(mode == 2)
-			point_at(target)
-		to_chat(usr, "<span class='notice'>You activate the pinpointer.</span>")
-	else
-		active = 0
-		icon_state = icon_off
-		to_chat(usr, "<span class='notice'>You deactivate the pinpointer.</span>")
+/obj/item/pinpointer/advpinpointer/process()
+	if(setting == 0)
```

Comment:
Can you make these into defines as well? Reading 0-2 doesn't tell much

Reply:
Defined